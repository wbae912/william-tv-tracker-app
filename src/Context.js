import React from 'react';

const tvContext = React.createContext({
  shows: [],
  deleteTvShow: () => {},
  addTvShow: () => {}
})

export default tvContext;