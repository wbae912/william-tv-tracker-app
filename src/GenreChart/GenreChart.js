import React, { Component } from 'react';
import { Bar } from 'react-chartjs-2';
import tvContext from '../Context';

export default class GenreChart extends Component {
  static contextType = tvContext;

  render() {
    const noGenre = this.context.shows.filter(show => show.genre === 'N/A').length;
    const countAction = this.context.shows.filter(show => show.genre === 'Action').length;
    const countAnimated = this.context.shows.filter(show => show.genre === 'Animated').length;
    const countComedy = this.context.shows.filter(show => show.genre === 'Comedy').length;
    const countDocumentary = this.context.shows.filter(show => show.genre === 'Documentary').length;
    const countDrama = this.context.shows.filter(show => show.genre === 'Drama').length;
    const countEducational = this.context.shows.filter(show => show.genre === 'Educational').length;
    const countFantasy = this.context.shows.filter(show => show.genre === 'Fantasy').length;
    const countHorror = this.context.shows.filter(show => show.genre === 'Horror').length;
    const countMystery = this.context.shows.filter(show => show.genre === 'Mystery').length;
    const countReality = this.context.shows.filter(show => show.genre === 'Reality').length;
    const countSciFi = this.context.shows.filter(show => show.genre === 'Sci-Fi').length;
    const countSitcom = this.context.shows.filter(show => show.genre === 'Sitcom').length;
    const countThriller = this.context.shows.filter(show => show.genre === 'Thriller').length;
    const countVariety = this.context.shows.filter(show => show.genre === 'Variety').length;
    const data = {
      labels: [
        'N/A',
        'Action',
        'Animated',
        'Comedy',
        'Documentary',
        'Drama',
        'Educational',
        'Fantasy',
        'Horror',
        'Mystery',
        'Reality',
        'Sci-Fi',
        'Sitcom',
        'Thriller',
        'Variety',
      ],
      datasets: [
        {
          label: 'TV Shows per Genre',
          backgroundColor: 'rgba(59, 45, 255, 0.2)',
          borderColor: 'rgba(59, 45, 255, 0.2)',
          borderWidth: 1,
          hoverBackgroundColor: 'rgba(255,99,132,0.4)',
          hoverBorderColor: 'rgba(255,99,132,1)',
          data: [noGenre, countAction, countAnimated, countComedy, countDocumentary, countDrama, countEducational, countFantasy,
            countHorror, countMystery, countReality, countSciFi, countSitcom, countThriller, countVariety]
        }
      ]
    };
    return (
      <div style={{width: '90%'}}>
        <Bar
          data={data}
          redraw
          width={100}
          height={250}
          options={{
            responsive: true,
            maintainAspectRatio: false,
            scales: {
              xAxes: [{
                  ticks: { display: true, fontColor: 'white' },
                  gridLines: {
                      display: false,
                      drawBorder: true
                  }
              }],
              yAxes: [{
                  ticks: { display: true, stepSize: 1, beginAtZero: true, min: 0, fontColor: 'white' },
                  gridLines: {
                      display: true,
                      drawBorder: true
                  }
              }]
          }
          }}
        />
      </div>
    )
  }
}