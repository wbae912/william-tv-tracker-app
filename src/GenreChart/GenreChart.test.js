import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Context from '../Context';
import GenreChart from './GenreChart';

describe(`Completed component`, () => {
  const shows = [
      {
        "id": "1",
        "tv_title": "Example",
        "status": "Completed",
        "season_number": "1",
        "episode_number": "1",
        "rating": "3",
        "genre": "Action",
        "description": "Lorem ipsum",
        "review": "Lorem ipsum"
      },
      {
        "id": "2",
        "tv_title": "Example 2",
        "status": "Completed",
        "season_number": "1",
        "episode_number": "1",
        "rating": "3",
        "genre": "Action",
        "description": "Lorem ipsum",
        "review": "Lorem ipsum"
      }
    ];

    const data = {
      labels: [
        'N/A',
        'Action',
        'Animated',
        'Comedy',
        'Documentary',
        'Drama',
        'Educational',
        'Fantasy',
        'Horror',
        'Mystery',
        'Reality',
        'Sci-Fi',
        'Sitcom',
        'Thriller',
        'Variety',
      ],
      datasets: [
        {
          label: 'TV Shows per Genre',
          backgroundColor: 'rgba(59, 45, 255, 0.2)',
          borderColor: 'rgba(59, 45, 255, 0.2)',
          borderWidth: 1,
          hoverBackgroundColor: 'rgba(255,99,132,0.4)',
          hoverBorderColor: 'rgba(255,99,132,1)',
          data: [1, 1, 1, 1, 1, 1, 1, 1,
            1, 1, 1, 1, 1, 1, 1]
        }
      ]
    };
  
  it('renders component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Context.Provider value={{shows}}><GenreChart /></Context.Provider>
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });
  // it('renders UI as expected', () => {
  //   const tree = renderer
  //     .create(<BrowserRouter><Context.Provider value={{shows}}><GenreChart data={data}/></Context.Provider></BrowserRouter>)
  //     .toJSON();
  //     expect(tree).toMatchSnapshot();
  // })
  // it('renders a .GenreChart by default', () => {
  //   const wrapper = shallow(<Context.Provider value={{shows}}><GenreChart /></Context.Provider>)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })
})