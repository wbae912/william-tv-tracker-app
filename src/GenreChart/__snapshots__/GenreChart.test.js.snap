// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreChart component renders a .GenreChart by default 1`] = `<GenreChart />`;
